libres/lib/analysis/update.cpp

```diff
@@ -684,22 +690,20 @@ void copy_parameters(enkf_fs_type *source_fs, enkf_fs_type *target_fs,
    // Do something with the pressure nodes ...
 
    free(pressure_nodes);
-
 */
-
-vector_type *alloc_node_ensemble(const ensemble_config_type *ens_config,
-                                 const int ens_size, enkf_fs_type *src_fs,
-                                 const char *key, int report_step) {
-    vector_type *node_ensemble = vector_alloc_new();
+std::vector<node> alloc_node_ensemble(const ensemble_config_type *ens_config,
+                                      const int ens_size, enkf_fs_type *src_fs,
+                                      const char *key, int report_step) {
+    std::vector<node> node_ensemble;
     const enkf_config_node_type *config_node =
         ensemble_config_get_node(ens_config, key);
     node_id_type node_id = {.report_step = report_step, .iens = -1};
 
     for (int iens = 0; iens < ens_size; iens++) {
-        enkf_node_type *node = enkf_node_alloc(config_node);
+        node n = node(enkf_node_alloc(config_node), enkf_node_free__);
```

Comment:
If you do the `enkf_node_deleter` suggestion, you don't need to pass a function:

```suggestion
        node n = enkf_node_alloc(config_node);
```

Reply:
I believe this will have to be `node n { enkf_node_alloc(config_node) }` or am I missing something?

Reply:
So I couldn't get the compiler to accept `decltype(enkf_node_deleter)`as an invocable type in a way that it would travel with unique_ptr so I made it a struct instead. Might be that it requires a more up to date c++ standard. I added a small refactor which should make things more clear.